{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f9bf1414",
   "metadata": {},
   "outputs": [],
   "source": [
    "# !pip install --no-cache-dir fdc-connector-python-sdk==1.0.6 --extra-index-url https://__token__:glpat-ZjxNEisHtmaFhxnYefd8@gitlab.fosfor.com/api/v4/projects/283/packages/pypi/simple"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "7c559c66",
   "metadata": {},
   "outputs": [
    {
     "ename": "ModuleNotFoundError",
     "evalue": "No module named 'fdc_connector_python_sdk'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[4], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mfdc_connector_python_sdk\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mfactory\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mconnectorFactory\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m ConnectionFactory\n",
      "\u001b[0;31mModuleNotFoundError\u001b[0m: No module named 'fdc_connector_python_sdk'"
     ]
    }
   ],
   "source": [
    "from fdc_connector_python_sdk.factory.connectorFactory import ConnectionFactory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a91fc0d7",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'ConnectionFactory' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[2], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m snowflake_connector \u001b[38;5;241m=\u001b[39m \u001b[43mConnectionFactory\u001b[49m\u001b[38;5;241m.\u001b[39mget_connector(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124msnowflake\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m      2\u001b[0m snowflake_dto \u001b[38;5;241m=\u001b[39m ConnectionFactory\u001b[38;5;241m.\u001b[39mget_dto(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mSNOWFLAKE\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "\u001b[0;31mNameError\u001b[0m: name 'ConnectionFactory' is not defined"
     ]
    }
   ],
   "source": [
    "snowflake_connector = ConnectionFactory.get_connector(\"snowflake\")\n",
    "snowflake_dto = ConnectionFactory.get_dto(\"SNOWFLAKE\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9533819",
   "metadata": {},
   "outputs": [],
   "source": [
    "param_dict = snowflake_dto.set_param(auth_type=auth_type, **connection_params)\n",
    "session = snowflake_connector.get_snowpark_session(param_dict)\n",
    "print(session)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
